File: frontend/src/microservices/operations/overall/objects/ManagerTaskListResponse.ts
// 鏈枃浠跺畾涔?operations 妯″潡鐨?`ManagerTaskListResponse`锛屼綔涓哄垪琛ㄥ搷搴旀暟鎹苟鎻愪緵 JSON 缂栬В鐮併€?

import type { ManagerTaskResponse } from './ManagerTaskResponse'

export type ManagerTaskListResponse = {
  tasks: ManagerTaskResponse[]
}
export const managerTaskListResponseFromJson = (json: string): ManagerTaskListResponse =>
  JSON.parse(json) as ManagerTaskListResponse

export const managerTaskListResponseToJson = (value: ManagerTaskListResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/ManagerTaskListResponse.scala
// 本文件定义 `ListManagerTasksPlanner` 返回的任务列表对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerTaskListResponse(tasks: List[ManagerTaskResponse])
object ManagerTaskListResponse:
  given sourceEncoder: Encoder[ManagerTaskListResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerTaskListResponse] = deriveDecoder
